### DISCUSSION TITLE
How can I feed DocBook content into Docusaurus?

### DISCUSSION THREAD

**Discussion by tomschr** [facebook/docusaurus#8177] [Q&A]
Hi Docusaurus team, :wave: 

such a cool project! :+1:

As far as I know, you only support Markdown/MDX as an input format. We have content in DocBook5 XML and want to feed this content into Docusaurus somehow. I see two possible ways:

* Transform the DocBook content into (simple?) HTML and feed that into Docusaurus somehow (is this possible?)
* Transform the DocBook content into Markdown/MDX with a custom transformation (I would have to write that). This option would loose semantic.

I suppose, you don't want to support DocBook as an input format. :wink: In the issue tracker, I saw #3845 for ASCIIDoc support which would be another way, however that is closed.

I'm searching for a way which can be automated and repeatedly done.

What other options do you see? Many thanks!

**Comment by slorber** [comment]
You assume I know DocBook but it's not really the case 😅 

We don't support any kind of XML file as a source format, but you can build a plugin to support it.

If you want to reuse existing plugins like docs/blog, you'd have to transform your XML to md as an intermediate step. Many people did that already (for example they generate md docs at build time based on some data source - cms, api ref, javadoc, AST, whatever...). 

We don't plan to provide such official support so you would have to build this yourself as a plugin or pre-build step

**Reply by tomschr** [reply]
Thanks you @slorber for your input. Much appreciated.

Well, I'm certainly not suggesting to support DocBook for Docusaurus. :wink: Just would like to know a way how I can feed Docusaurus, be it DocBook, ASCIIDoc, or any other format.

Is there a direct way to feed Docusaurus with HTML, without the intermediate step of Markdown/MDX?

You mentioned a plugin. Is there any documentation about how to write plugins?

**Reply by slorber** [reply]
you can put html content in `/static` and embed it inside docs using iframes in MDX maybe, or just linking to it

Yes there is documentation for how to write a plugin

**Reply by tomschr** [reply]
> Yes there is documentation for how to write a plugin

Well... I suppose you mean this https://docusaurus.io/docs/advanced/plugins ?

Probably that's beyond my abilities. I'm not a JavaScript expert, at least not now. This is a bit too general, I need an example to connect the pieces, or learn from a tutorial. Do you have any links like that?

Thanks.

**Reply by slorber** [reply]
You need skills to achieve what you want. There is no shortcut and you have to do the work.
